analyzers/src/SonarAnalyzer.CSharp/Rules/VariableShadowsField.cs

```diff
@@ -55,70 +55,56 @@ protected override void Initialize(SonarAnalysisContext context)
+                    ReportOnVariableMatchingField(c, members, declaration.Identifier);
                 },
                 SyntaxKind.ForEachStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((LocalDeclarationStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.LocalDeclarationStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDeclaration(c, ((LocalDeclarationStatementSyntax)c.Node).Declaration), SyntaxKind.LocalDeclarationStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDeclaration(c, ((ForStatementSyntax)c.Node).Declaration), SyntaxKind.ForStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDeclaration(c, ((UsingStatementSyntax)c.Node).Declaration), SyntaxKind.UsingStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDeclaration(c, ((FixedStatementSyntax)c.Node).Declaration), SyntaxKind.FixedStatement);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((ForStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.ForStatement);
-
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((UsingStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.UsingStatement);
-
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c => ProcessStatementWithVariableDeclaration((FixedStatementSyntax)c.Node, s => s.Declaration, c), SyntaxKind.FixedStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDesignation(c, ((DeclarationPatternSyntaxWrapper)c.Node).Designation), SyntaxKindEx.DeclarationPattern);
+            context.RegisterSyntaxNodeActionInNonGenerated(c => ProcessVariableDesignation(c, ((DeclarationExpressionSyntaxWrapper)c.Node).Designation), SyntaxKindEx.DeclarationExpression);
         }
 
-        private static void ProcessStatementWithVariableDeclaration<T>(T declaration, Func<T, VariableDeclarationSyntax> variableSelector, SyntaxNodeAnalysisContext context)
+        private static void ProcessVariableDesignation(SyntaxNodeAnalysisContext context, VariableDesignationSyntaxWrapper variableDesignation)
         {
-            var variableDeclaration = variableSelector(declaration);
-            if (variableDeclaration == null)
-            {
-                return;
-            }
-
-            var variables = variableDeclaration.Variables;
-
-            List<ISymbol> members = null;
-            foreach (var variable in variables)
+            if (variableDesignation.AllVariables() is { Length: > 0 } variables
+                && context.SemanticModel.GetDeclaredSymbol(variables[0]) is { ContainingType: { } containingType }
```

Comment:
silly question - can `context.ContainingSymbol.ContainingType ` be used instead of doing a `GetDeclaredSymbol` call?

same for ProcessVariableDeclaration